autowiring/DecorationDisposition.h

```diff
@@ -103,24 +92,36 @@ struct DecorationDisposition
       Multi
     };
 
-    Subscriber(void) {}
-    Subscriber(bool is_shared, Type type, SatCounter* satCounter) :
+    Subscriber(void) = default;
+
+    Subscriber(bool is_shared, Type type, autowiring::altitude altitude, SatCounter* satCounter) :
       is_shared{is_shared},
       type{type},
+      altitude{altitude},
       satCounter{satCounter}
-    {}
+    {
+      if (satCounter == nullptr)
+        throw std::runtime_error("Cannot initialize Subscriber with nullptr to SatCounter.");
+    }
 
     // True if a shared pointer will be taken, false otherwise
-    bool is_shared = false;
+    const bool is_shared;
 
     // The relationship between this subscriber and the provided decoration
-    Type type = Type::Normal;
+    const Type type;
+
+    // The altitude of the satisfaction counter
+    const autowiring::altitude altitude;
 
-    // The satisfaction counter itself
-    SatCounter* satCounter = nullptr;
+    // The pointer to the satisfaction counter, it should never be nullptr
+    SatCounter* const satCounter;
+
+    bool operator<(const Subscriber& rhs) const {
+      return std::tie(altitude, satCounter) > std::tie(rhs.altitude, rhs.satCounter);
+    }
   };
 
-  std::vector<Subscriber> m_subscribers;
+  std::set<Subscriber> m_subscribers;
```

Comment:
I'm reluctant to make this kind of change.  I understand that it has the advantage of making it easy to prevent reduplicated introductions, but that kind of guarantee should be made higher up, not here.